XXX-buffer-alloc-from.md

```diff
@@ -0,0 +1,50 @@
+# Buffer.alloc / Buffer.allocUnsafe / Buffer.from
+
+Status | Draft
+------ | ----
+Author | James M Snell
+Date   | 25 Jan 2016
+
+## Purpose
+
+* Soft-Deprecate the existing `Buffer()` constructors and introduce new
+  `Buffer.from()`, `Buffer.alloc()` and `Buffer.allocUnsafe()` factory methods
+  to address Buffer API usability and security concerns.
+* Add `--zero-fill-buffers` command line option to node to force all Buffers
+  to zero-fill when created
+
+## Details
+
+The details of the proposal are captured by [Pull Request 4682][] in the
+nodejs/node repository (https://github.com/jasnell/node/tree/buffer-safe).
+
+### Buffer.alloc(size[, fill])
+
+Allocates a new initialized buffer of `size` bytes. If `fill` is specified
+and is a string, the Buffer is allocated off the shared pool by calling:
```

Comment:
@trevnorris ... +1 SGTM
On Jan 25, 2016 4:27 PM, "Trevor Norris" <notifications@github.com> wrote:

> In XXX-buffer-alloc-from.md
> <https://github.com/nodejs/node-eps/pull/7#discussion_r50779681>:
>
> > +
> > +* Soft-Deprecate the existing `Buffer()` constructors and introduce new
> > +  `Buffer.from()`, `Buffer.alloc()` and `Buffer.allocUnsafe()` factory methods
> > +  to address Buffer API usability and security concerns.
> > +* Add `--zero-fill-buffers` command line option to node to force all Buffers
> > +  to zero-fill when created
> > +
> > +## Details
> > +
> > +The details of the proposal are captured by [Pull Request 4682][] in the
> > +nodejs/node repository (https://github.com/jasnell/node/tree/buffer-safe).
> > +
> > +### Buffer.alloc(size[, fill])
> > +
> > +Allocates a new initialized buffer of `size` bytes. If `fill` is specified
> > +and is a string, the Buffer is allocated off the shared pool by calling:
>
> Also, is Buffer.alloc(1e8) really allocated from the pool?
>
> Allocations are of size Buffer.poolSize >> 1 or smaller. So it could,
> depending on what the user sets poolSize to. But by default no.
>
> @jasnell <https://github.com/jasnell> Since we're here, may want to add Buffer.alloc(size[,
> fill[, encoding]]) so the encoding of the string can be specified. I
> should have done this on initial implementation but didn't think about it
> then.
>
> Also, if alloc() is used then I suggest we also don't use a pool. Since
> other Buffer's contents can be accessed via buffer.buffer it might be
> considered a security violation, and the whole point of this thing is to
> not do that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nodejs/node-eps/pull/7/files#r50779681>.
>